src/docfx/lib/JsonUtility.cs

```diff
@@ -181,12 +203,75 @@ private static void Traverse(this JToken token, List<Error> errors, List<JToken>
                     }
                     else
                     {
-                        prop.Value.Traverse(errors, nullNodes, prop.Name);
+                        prop.Value.TraverseForNullValueValidation(errors, nullNodes, prop.Name);
                     }
                 }
             }
         }
 
+        private static void TraverseForMismatchingTypeValidation(this JToken token, List<Error> errors, Type type)
+        {
+            if (token is JArray array)
+            {
+                foreach (var item in token.Children())
+                {
+                    item.TraverseForMismatchingTypeValidation(errors, type);
+                }
+            }
+            else if (token is JObject obj)
+            {
+                foreach (var item in token.Children())
+                {
+                    var prop = item as JProperty;
+                    Type nestedType;
+                    (nestedType, errors) = CheckFieldTypeNotExistingInSchema(type, prop, errors);
+                    prop.Value.TraverseForMismatchingTypeValidation(errors, nestedType);
+                }
+            }
+        }
+
+        private static (Type, List<Error>) CheckFieldTypeNotExistingInSchema(Type type, JProperty prop, List<Error> errors)
```

Comment:
>CheckFieldTypeNotExistingInSchema [](start = 43, length = 33)

If the type has a member with [JsonExtensionData], we should not warn

Reply:
Forgotten, will add a test for it

---
In reply to: [205663105](https://github.com/dotnet/docfx/pull/3112#discussion_r205663105) [](ancestors = 205663105)